File: Source/Game/Rounds/RoundUtilities.cs
using WCSharp.Api;
using static WCSharp.Api.Common;

public static class RoundUtilities
{
    public static void MovePlayerToStart(player Player)
    {
        var kitty = Globals.ALL_KITTIES[Player];
        var x = RegionList.SpawnRegions[Player.Id].Center.X;
        var y = RegionList.SpawnRegions[Player.Id].Center.Y;
        kitty.Unit.X = x;
        kitty.Unit.Y = y;
        kitty.Unit.Facing = 359.000f;
    }

    public static void MoveTeamToStart(Team team)
    {
        for (int i = 0; i < team.Teammembers.Count; i++)
        {
            var kitty = team.Teammembers[i];
            kitty.Finished = true;
            MovePlayerToStart(kitty.Player);
        }
        team.Finished = true;
    }

    public static void MoveAllPlayersToStart()
    {
        foreach (var kitty in Globals.ALL_KITTIES_LIST)
        {
            MovePlayerToStart(kitty.Player);
        }
    }

    public static void RoundResetAll()
    {
        foreach (var kitty in Globals.ALL_KITTIES_LIST)
        {
            kitty.Unit.Revive(RegionList.SpawnRegions[kitty.Player.Id].Center.X, RegionList.SpawnRegions[kitty.Player.Id].Center.Y, false);
            kitty.Circle.HideCircle();
            kitty.Alive = true;
            kitty.ProgressZone = 0;
            kitty.Finished = false;
            kitty.Unit.Mana = kitty.Unit.MaxMana;
            kitty.CurrentStats.ResetRoundData();
        }
    }

    public static void MovedTimedCameraToStart()
    {
        var x = RegionList.SpawnRegions[0].Center.X;
        var y = RegionList.SpawnRegions[0].Center.Y;
        foreach (var player in Globals.ALL_PLAYERS)
        {
            if (player.IsLocal) PanCameraToTimed(x, y, RoundManager.END_ROUND_DELAY);
            CameraUtil.RelockCamera(player);
        }
    }
}

File: src/Game/Rounds/RoundUtilities.ts
import { Team } from 'src/Gamemodes/Teams/Team'
import { Globals } from 'src/Global/Globals'
import { RegionList } from 'src/Global/RegionList'
import { CameraUtil } from 'src/Utility/CameraUtil'
import { MapPlayer } from 'w3ts'

export class RoundUtilities {
    public static MovePlayerToStart(Player: MapPlayer) {
        const kitty = Globals.ALL_KITTIES.get(Player)!
        const x = RegionList.SpawnRegions[Player.id].centerX
        const y = RegionList.SpawnRegions[Player.id].centerY
        kitty.Unit.setPosition(x, y)
        kitty.Unit.facing = 360.0
    }

    public static MoveTeamToStart(team: Team) {
        for (let i = 0; i < team.Teammembers.length; i++) {
            const player = team.Teammembers[i]
            const kitty = (Globals.ALL_KITTIES.get(player)!.Finished = true)
            RoundUtilities.MovePlayerToStart(player)
        }
        team.Finished = true
    }

    public static MoveAllPlayersToStart = () => {
        for (const [_, kitty] of Globals.ALL_KITTIES) {
            RoundUtilities.MovePlayerToStart(kitty.Player)
        }
    }

    public static RoundResetAll = () => {
        for (const [_, kitty] of Globals.ALL_KITTIES) {
            kitty.Unit.revive(
                RegionList.SpawnRegions[kitty.Player.id].centerX,
                RegionList.SpawnRegions[kitty.Player.id].centerY,
                false
            )
            Globals.ALL_CIRCLES.get(kitty.Player)?.HideCircle()
            kitty.Alive = true
            kitty.ProgressZone = 0
            kitty.Finished = false
            kitty.Unit.mana = kitty.Unit.maxMana
            kitty.CurrentStats.ResetRoundData()
        }
    }

    public static MovedTimedCameraToStart = () => {
        const x = RegionList.SpawnRegions[0].centerX
        const y = RegionList.SpawnRegions[0].centerY
        for (const player of Globals.ALL_PLAYERS) {
            if (player.isLocal()) PanCameraToTimed(x, y, Globals.END_ROUND_DELAY)
            CameraUtil.RelockCamera(player)
        }
    }
}
